### PR TITLE
Add 'shallow_clone' option (default: true)

### PR DESCRIPTION
**Pull request by floriankrueger**
a possible fix for #21

### CODE REVIEW

**Comment by KrauseFx** [comment]
Thanks for the PR. Don't worry about the broken build. Is there any reason why this shouldn't be default behaviour?

**Comment by KrauseFx** [comment]
Why not just always use shallow?

**Comment by KrauseFx** [comment]
What I mean, we can probably just remove the option, right? And just have it shallow always

**Comment by floriankrueger** [comment]
actually as far as I can see 'shallow_clone' would be switched on by default (as it is the current behaviour).

The reason we seem to need the possibility to switch to full clones (so no --depth=1 parameter) is that some git servers (afaik servers with git < 1.9) do not allow pushes from shallow clones. 

```
fatal: protocol error: expected old/new/ref, got 'shallow edeb3181db739fd366...'
```

That's what I expected on an internal [Stash](https://www.atlassian.com/software/bitbucket/server) server. And it seems to be an issue for more users than just me (see #21).

**Comment by KrauseFx** [comment]
Very good point, thanks @floriankrueger :+1:

**Comment by KrauseFx** [comment]
This will be in the next release :rocket:

File: lib/match/git_helper.rb [hunk 1]
```diff
@@ -1,10 +1,12 @@
 module Match
   class GitHelper
-    def self.clone(git_url)
+    def self.clone(git_url, shallow_clone)
```
Comment (lacostej) [inline lib/match/git_helper.rb]: What's the recommended code style to add parameters to existing methods ?

Maybe ?

``def self.clone(git_url, shallow_clone: true)``

Comment (KrauseFx) [inline lib/match/git_helper.rb]: Very good point. Does this PR break anything? Should we change it?

Comment (floriankrueger) [inline lib/match/git_helper.rb]: Oh sorry I didn't know about any code style guidelines. If you want to, I'm happy to update my pull request.

**Pull request merged**